File: src/algorithms/recursion/nth-fibonacci/solution1.py
# STRATEGY:
# Naive recursive solution
# COMPLEXITY:
# O(2^n) Time | O(n) Space

def getNthFib(n):
    if n == 2:
    	return 1
    elif n == 1:
    	return 0
    else:
    	return getNthFib(n - 1) + getNthFib(n - 2) 

File: src/algorithms/recursion/nth-fibonacci/solution1.ts
// O(2*n) Time | O(n) Space
export function getNthFib(n: number): number {
  if (n === 2) {
    return 1;
  } else if (n === 1) {
    return 0;
  } else {
    return getNthFib(n - 1) + getNthFib(n - 2);
  }
}
